src/js/pyodide.ts

```diff
@@ -219,7 +252,19 @@ export async function loadPyodide(
   if (!indexURL.endsWith("/")) {
     indexURL += "/";
   }
-  options.indexURL = indexURL;
+  const options_ = options as ConfigType;
+  if (!options.lockFileContents) {
+    const lockFileURL = options.lockFileURL ?? indexURL + "pyodide-lock.json";
+    options_.lockFileContents = loadLockFile(lockFileURL);
+    // packageBaseUrl isn't present, try using base location of lockFileUrl. If
+    // lockFileURL is relative, use location as the base URL.
+    options_.packageBaseUrl ??= calculateInstallBaseUrl(lockFileURL);
+  }
+  options_.indexURL = indexURL;
+  // cdnUrl only for node.
+  options_.cdnUrl =
+    options_.packageBaseUrl ??
+    `https://cdn.jsdelivr.net/pyodide/v${version}/full/`;
```

Comment:
Shouldn't it always be `https://cdn.jsdelivr.net/pyodide/v${version}/full/`, as we use this only for a fallback?